vm/src/obj/objbytes.rs

```diff
@@ -420,6 +422,117 @@ impl PyBytesRef {
     fn rmul(self, n: PyIntRef, vm: &VirtualMachine) -> PyResult {
         self.repeat(n, vm)
     }
+
+    /// Return a string decoded from the given bytes.
+    /// Default encoding is 'utf-8'.
+    /// Default errors is 'strict', meaning that encoding errors raise a UnicodeError.
+    /// Other possible values are 'ignore', 'replace'
+    /// For a list of possible encodings,
+    /// see https://docs.python.org/3/library/codecs.html#standard-encodings
+    /// currently, only 'utf-8' and 'ascii' emplemented
+    #[pymethod(name = "decode")]
+    fn decode(
+        self,
+        encoding: OptionalArg<PyStringRef>,
+        errors: OptionalArg<PyStringRef>,
+        vm: &VirtualMachine,
+    ) -> PyResult<String> {
+        let mut strict_mod = true;
+        let replacing_char = match errors {
+            OptionalArg::Present(ref input) => match input.as_str() {
+                "replace" => {
+                    strict_mod = false;
+                    Some('\u{FFFD}')
+                }
+                "ignore" => {
+                    strict_mod = false;
+                    None
+                }
+                _ => None,
+            },
+            OptionalArg::Missing => None,
+        };
+        let encoding_type = match encoding {
+            OptionalArg::Present(ref input) => input.as_str(),
+            OptionalArg::Missing => "utf-8",
+        };
+
+        let decode_error = Err(vm.new_value_error("DecodeError".to_string()));
+
+        let mut decode_content = String::new();
+        match encoding_type {
+            "ascii" => {
+                for &b in self.get_value() {
+                    match b.is_ascii() {
+                        true => decode_content.push(b as char),
+                        false => {
+                            if !strict_mod && replacing_char.is_some() {
+                                decode_content.push(replacing_char.unwrap())
+                            }
+                        }
+                    }
+                }
+            }
+            // utf-8 is default encoding
+            _ => {
+                let mut p: u32 = 0u32;
+                let mut remaining_bytes = 0;
```

Comment:
utf-8 is the default, but cpython throws a `LookupError` if you give it an invalid encoding.

Reply:
Got it, I will implement this.

Reply:
Done :)